kubetest2-gke/deployer/up.go

```diff
@@ -105,22 +166,19 @@ func (d *deployer) Up() error {
 				args = append(args, subNetworkArgs...)
 				args = append(args, privateClusterArgs...)
 				args = append(args, cluster.name)
-				if err := runWithOutput(exec.CommandContext(ctx, "gcloud", args...)); err != nil {
+				var buf bytes.Buffer
+				if err := runWithOutputAndBuffer(exec.CommandContext(ctx, "gcloud", args...), &buf); err != nil {
 					// Cancel the context to kill other cluster creation processes if any error happens.
 					cancel()
-					return fmt.Errorf("error creating cluster: %v", err)
+					return fmt.Errorf("error creating cluster %q: %w", cluster.name, errors.New(buf.String()))
```

Comment:
won't this buffer also have stdout ?

Reply:
Yeah the output will be written to both stdout and the buffer.

The reasons I'm writing like this are:
1. We need to capture the output for matching the error pattern to decide if it needs to be retried
2. We still need to keep the stdout and stderr for informational purpose

Does it make sense to you?